### PR TITLE
Add notifications

### PR DESCRIPTION
**Pull request by rizchelwood**
Screen Shot 2019-09-18 at 2 25 42 PM (2)

### CODE REVIEW

**Comment by rizchelwood** [comment]
@greglobinski have you seen these errors when running `gatsby-interface`? I know a lot of stuff has been added so wondering if you've seen this 

When installing the package in a repo:
```
/node_modules/gatsby/cache-dir/gatsby-browser-entry.js:25:4: Unexpected token (25:4)
  23 |
  24 |   return (
> 25 |     <React.Fragment>
     |    ^
  26 |       {finalData && render(finalData)}
  27 |       {!finalData && <div>Loading (StaticQuery)</div>}
  28 |     </React.Fragment>
```

When linking the package:
```
/node_modules/gatsby/cache-dir/gatsby-browser-entry.js:17:2: Unexpected token (17:2)
  15 
const StaticQuery = props => (
  <StaticQueryContext.Consumer>

   {staticQueryData => {
       if (
        props.data ||
```

**Comment by rizchelwood** [comment]
Problem: The `Link` component will not work with sites that don't run any Gatsby commands to compile the link directive 

Resolution - to make the library more agnostic, we're going to start passing in Gatsby `Link` to components where it's needed.

File: src/components/Link/Link.js [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
 import { colors } from "../../utils/presets"
 
-const Link = ({ children, href, target, to, variant = `DEFAULT` }) => {
+const Link = ({
+  children,
+  href,
+  target,
+  to,
+  variant = `DEFAULT`,
+  LinkComponent,
```
Comment (DSchau) [inline src/components/Link/Link.js]: Clever solution!

**Comment by greglobinski** [comment]
@rizchelwood  I'm not sure we should go this way. 

> to make the library more agnostic

do we really want to create platform agnostic lib, I think we create `GATSBY-interface` not `UNIVERSAL-interface`. The solution with `LinkComponent` works pretty well when it's hidden inside `Notification`, but using the `Link` component as a standalone component in the app with setting `LinkComponen` prop every time the component will link to inner page will be pain in the ass. 

I guess we will have similar problem when we will want to use `gatsby-image` inside `Avatar` component. 

The 'agnostic' way is closed for us since we decided to use `emotion` for styling, in a moment  we will use `formik` to manage forms, in the near future `react-apollo` :D etc

and I'm ok with that  - we create `gatsby-interface` to work on `gatsby` apps/sites.

That's my thinking

cc/ @DSchau

**Comment by DSchau** [comment]
> I guess we will have similar problem when we will want to use gatsby-image inside Avatar component.

gatsby-image doesn't depend upon Gatsby like the Link component does, but I get your general comment here.

> do we really want to create platform agnostic lib, I think we create GATSBY-interface not UNIVERSAL-interface. The solution with LinkComponent works pretty well when it's hidden inside Notification, but using the Link component as a standalone component in the app with setting LinkComponen prop every time the component will link to inner page will be pain in the ass ;) and counter productive.

The best way would be to make it agnostic but _default_ it to the gatsby-link component. Unfortunately, that's not really possible since the import/require statement itself seems to cause the issue.

@rizchelwood could you share the gatsby-preview project you were using yesterday to test this? I'd like to take a look and mess around a bit!

**Comment by greglobinski** [comment]
> gatsby-image doesn't depend upon Gatsby like the Link component does,

Good to know, thank you Dustin

**Comment by greglobinski** [comment]
@DSchau 	

> could you share the gatsby-preview project you were using yesterday

there is something that I do not understand obviously, is not true that "gatsby-preview" sites must be gatsby-build sites?

**Comment by rizchelwood** [comment]
re: library more agnostic

@greglobinski I definitely see your point but if we want to use this library in extensions, 3rd party integrations, etc - it needs to be more agnostic. It would be a pain to copy and paste these components in every extension we do and it would make maintenance of these other apps harder.

I'm not sure if it's right to assume we'd use this only in `gatsby` apps since we want to use the same components in external apps (which the setup in some cases are out of our control). 

@DSchau it's here - https://github.com/gatsby-inc/extensions. Parcel doesn't play nice with emotion it seems so for now I've just copied the `Notification` component

**Comment by rizchelwood** [comment]
@greglobinski Ultimately we need to make the decision if we're going to support non Gatsby apps or only Gatsby apps. I think there's pros and cons to both so we need to decide which outweighs the other. This will also help us with making future decisions :)

**Comment by greglobinski** [comment]
ok, now I get it - thank you :)

so we have this issue only with `gatsby-link` so my thinkin is change the name of the current `Link` to `BaseLink` and above it create a new `Link` which will have a default value of `LinkComponent,` set to `gatsby-link`, so in plugins we will use `BaseLink` but in the cloud app we will use `Link`.

> Ultimately we need to make the decision if we're going to support non Gatsby apps or only Gatsby apps

before your comment I would said "do not care about non Gatsby apps" but I completely forgot about the plugins, sorry!

**Comment by rizchelwood** [comment]
> change the name of the current Link to BaseLink and above it create a new Link which will have a default value of LinkComponent set to gatsby-link, so in plugins we will use BaseLink but in the cloud app we will use Link. 

We won't be able to have `gatsby-link` set anywhere in `gatsby-interface` because it will still get pulled in from the package and break non-Gatsby apps

**Comment by rizchelwood** [comment]
@greglobinski this is ready for review. I've removed the link stuff for now and added the close functionality

File: src/components/Notification/Notification.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  const Icon = content && NOTIFICATION_ICONS[tone]
+
+  if (!showNotificationState) return null
+
+  return (
+    <Component
+      css={{
+        ...styles.base,
+        ...styles.variants[variant](tones[tone]),
+        ...css,
```
[the remark this answers is not in the document]
Comment (rizchelwood) [reply src/components/Notification/Notification.js]: Will make this change 👍

**Comment by rizchelwood** [comment]
re: padding on secondary and primary - that's what I thought too but when you look in the code and what's out in prod, the padding is not the same

**Comment by greglobinski** [comment]
@rizchelwood Sorry, I only took a look at the storybook

**Pull request merged**